### DISCUSSION TITLE
[auth] Invalid copilot token: missing token: 403

### DISCUSSION THREAD

**Discussion by CafuChino** [community/community#114487] [Copilot Conversations] [answered]
### Select Topic Area

Bug

### Body

Issue Description: Encountered error message "[auth] Invalid copilot token: missing token: 403" in VSCode Copilot output log, and "✗ Error: No active Copilot license detected" in Copilot CLI. 

However, I have already subscribed to the yearly service.

I have tried:

- Try to logout and re-login
- Different ide s
- Purchase again (can not work)
- open a support ticket (no response)

**Comment by Akash1134** [comment]
Hi @CafuChino ,

 If you are a Copilot Individual user and getting the error using Copilot while trying to sign in, please contact our Support team for checking on your account.

Are you only getting the `missing token: error 403` while using Copilot CLI? If you are a Copilot for Business user authenticated to use Copilot via your Organization and **only getting this error while using Copilot CLI**, please check for an email sent to your team on March 14th with the subject line Subject: GitHub Copilot in the CLI will be generally available on March 21.

- If you do not have a billing option attached to your GitHub Copilot subscription you will need to set that up before following the below instructions.
- In order to resolve this issue you should go to your [Enterprise Copilot settings](https://github.com/enterprises/ENTERPRISE_NAME/settings/copilot?tab=policies) Or at the [Organization level](https://github.com/organizations/ORG_NAME/settings/copilot/policies) and you would need to change the option Copilot in the Cli to enabled. This would fix the issue at this time according to our Copilot engineering team.

Thanks!

**Comment by CafuChino** [comment, marked as answer]
Update (02/04/2024): According to @Unicorn-V, using copilot through a hosted VPS (Tencent Cloud) will result in your access being removed.

I encountered this issue a few days ago. According to the response from the ticket, the cause of the issue was that Github "perceived" an excessive or automated usage behavior from my end, leading to a unilateral ban on Copilot privileges without any notification or warning, even though I may have paid for this service. However, when I requested evidence to substantiate this action, my request was denied. Instead, I was repeatedly threatened with a permanent ban on my Github account if they still "perceive" similar behavior from me.

Therefore, I would like to alert those who have encountered this problem (or have yet to encounter it) to the following points:

Github may unilaterally suspend your account at any time.
Despite subscribing and paying for a feature you need, Github can still unilaterally disable this feature when they deem necessary, without providing any notification, explanation, or compensation.
Your data on Github is not secure, as Github can permanently ban your account when they deem necessary, especially when you seek to challenge their actions, they respond with threats.
Given these circumstances, I would advise, if possible, to consider other code completion tools, backup your data as much as possible, and turn to open-source solutions to minimize reliance on Github.

After this incident, I have lost confidence in Github's service level and data security. This is because, in the same ticket, I was threatened twice with a permanent account ban, when all I wanted to know was what I had done to warrant a ban on my paid subscription feature. All they did was issue threats and direct me to read the user agreement.

**Reply by bodhianhaha** [reply]
[[auth] Invalid copilot token: missing token: 403](https://github.com/orgs/community/discussions/114487#top)

**Reply by Akash1134** [reply]
Hi @aleksandr-vologodskij ,

Thanks for getting in touch, and sorry for the difficulty you're experiencing. I'm afraid that due to current trade controls, your GitHub account cannot make use of Copilot. Please refer this article to know more about the [GitHub and Trade Controls site policy](https://docs.github.com/en/site-policy/other-site-policies/github-and-trade-controls).

While GitHub obtained a license in 2021 to provide cloud services to Iranian users, Copilot is not included in the scope of this license.

I'm sorry if that wasn't what you were hoping to hear. We want every developer to be able to collaborate on GitHub, and are undertaking efforts to increase Copilot availability around the world, so please know that GitHub will continue to advocate for the right of developers in trade-restricted regions to access tools for code collaboration and communication.

**Comment by AboubakreSenouci** [comment]
#162

**Comment by Unicorn-V** [comment]
Has your access to github copilot been restored? I have the same problem as you. After I submitted a work order to the official, they told me that because a violation was detected, my access to copilot would not be restored for the time being. But in fact, I did nothing. What should I do?
![answer](https://github.com/community/community/assets/80693265/e8f13523-7ba7-4c92-9b3d-07d72cd772f9)

**Reply by CafuChino** [reply]
I'm sorry to hear about your situation; my usage permission has been restored. However, I am considering switching my code completion tool to another platform. Consider submitting a ticket again or temporarily using another tool. As I mentioned, if Github intentionally blocks your access, you will be helpless.

**Reply by Unicorn-V** [reply]
Thanks a lot, your advice is good. I have submitted a new ticket to explain my situation, and I will also look for new tools to help me learn coding.
Have a nice day : )
